app/perfaddresscache.h

```diff
@@ -42,17 +42,22 @@ class PerfAddressCache
 
     struct SymbolCacheEntry
     {
-        SymbolCacheEntry(quint64 offset = 0, quint64 size = 0, const QByteArray &symname = {})
+        SymbolCacheEntry(quint64 offset = 0, quint64 value = 0, quint64 size = 0, const QByteArray &symname = {}, const QByteArray &mangledSymName = {})
             : offset(offset)
+            , value(value)
             , size(size)
             , symname(symname)
+            , mangledSymName(mangledSymName)
         {}
 
         bool isValid() const { return !symname.isEmpty(); }
 
         quint64 offset;
+        // st_value in elf symbol table entry
+        quint64 value;
```

Comment:
And this is also still in, and neither the commit message nor a comment talks about it at all: Why do we need it? How is it different from offset?

Reply:
offset is equal to 'symAddr - elfStart' and value is sym.st_value - start of function.
They can be different. And they are absolutely different for LBR.

Reply:
That doesn't really clear things up for me. Both seem to indicate the start of the function for me. My understanding is:

- `symAddr` is the relocated address at which the function starts
- `elfStart` is the address where the ELF object got mapped to
- thus `offset` is the position independent start of the function that's stored in the symbol table
- what is `st_value` then, if it's also a start of the function? Which one of the two would be the correct address to pass as `--start-address` to `objdump`?

And I don't understand what LBR would have to do with this? The symbol table isn't affected by the perf sampling method after all - it's only affected by compiler options but is static afterwards...

Reply:
> * what is `st_value` then, if it's also a start of the function? Which one of the two would be the correct address to pass as `--start-address` to `objdump`?
>
    Proper address is sym.st_value to pass as `--start-address` to `objdump`.
> And I don't understand what LBR would have to do with this? The symbol table isn't affected by the perf sampling method after all - it's only affected by compiler options but is static afterwards...
    You can see that they are different for LBR on demo sample. It is pretty small and appropriate to check this.

Reply:
How do I check the different values? Can you please paste it? Also put the explanation of the difference into the comment then please

Reply:
I'm still waiting on a response to this thread

Reply:
You can record perf.data with LBR on 'demo' application and see under debugger at the break point inside cacheSymbols where 
cache.append({symAddr - elfStart, sym.st_value, sym.st_size, symbol}); 
You will see that 'symAddr - elfStart' differs from  sym.st_value.
I don't know how to explain. I just see that they are different. 
Ok, I will paste examples of values.

Reply:
ah, the difference is that offset is the "absolute" or "adjusted" value. So please put the following into the comments:

```
/// adjusted/absolute st_value, see documentation of the `addr` arg in `dwfl_module_getsym_info`
quint64 offset;
/// unadjusted/relative st_value
quint64 st_value;
```

that should make it clear what the difference between these two values is